src/Microsoft.Data.Analysis/DataFrame.IO.cs

```diff
@@ -454,24 +454,48 @@ public TextReader GetTextReader()
 
                     if (header)
                     {
-                        var headerColumns = string.Join(separator.ToString(), columnNames);
+                        bool firstColumn = true;
+                        var headerColumns = new StringBuilder();
+                        foreach (string name in columnNames)
+                        {
+                            if (!firstColumn)
+                            {
+                                headerColumns.Append(separator);
+                            }
+                            else
+                            {
+                                firstColumn = false;
+                            }
+
+                            bool needsQuotes = ((string)name).Contains(separator.ToString()) || ((string)name).Contains("\n");
```

Comment:
A couple things about this line:

1. Why the casts `(string)`? `name` is already declared as a string.
2. It is faster to check for `char` in a string than to check for `string`. And since netstandard2.0 doesn't have `string.Contains(char)`, you can instead say `name.IndexOf(separator) != -1 || name.IndexOf('\n') != -1`.
3. Calling `.ToString()` in a loop on something that doesn't change in the loop is a bad idea. It will allocate a new string every time through the loop.

Reply:
1. Good catch, copy paste error.
2. Yep, I tried Contains(char) and realized the same thing. That suggestion is a good workaround, thanks.
3. Good point, I'll keep that in mind for the future.